src/docs/guides/23-how-to-use-custom-pricing/index.md

```diff
@@ -0,0 +1,793 @@
+
+```js
+const bookingData =
+  startDate && endDate
+    ? {
+        unitType,
+        unitPrice,
+        startDate,
+        endDate,
+
+        // NOTE: If unitType is `line-item/units`, a new picker
+        // for the quantity should be added to the form.
+        quantity: 1,
+        cleaningFee: selectedCleaningFee,
+      }
+    : null;
+```
+
+Also add the following line inside the `BookingDatesForm` component
+before the `return` statement:
+
+```js
+const cleaningFeeLabel = intl.formatMessage({
+  id: 'BookingDatesForm.cleaningFee',
+});
+```
+
+Now we can add an input for selecting a cleaning fee to be added to a
+booking, for example right after the `FieldDateRangeInput` inside the
+`Form` element:
+
+```js
+{
+  cleaningFee ? (
+    <div className={css.cleaningFee}>
+      <FieldCheckbox
+        className={css.cleaningFeeLabel}
+        id={`${form}.cleaningFee`}
+        label={cleaningFeeLabel}
+        name={'additionalItems'}
+        value={'cleaningFee'}
+      />
+      <span className={css.cleaningFeeAmount}>
+        {formatMoney(intl, cleaningFee)}
+      </span>
+    </div>
+  ) : null;
+}
+```
+
+The input can be styled by adding the following style definitions to
+`BookingDatesForm.css`:
+
+```css
+.cleaningFee {
+  margin: 0 24px 38px 24px;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  flex-shrink: 0;
+
+  @media (--viewportMedium) {
+    margin: 0 0 38px 0;
+  }
+}
+
+.cleaningFeeAmount {
+  @apply --marketplaceListingAttributeFontStyles;
+  font-weight: 600;
+  margin: 0 0 0 10px;
+
+  @media (--viewportMedium) {
+    font-weight: 700;
+    margin: 0 0 0 10px;
+  }
+}
+```
+
+The next task is to modify the `EstimatedBreakdownMaybe` component to
+support the cleaning fee and add a new line item type to the booking
+breakdown. In the `EstimatedBreakdownMaybe` component:
+
+- Import `LINE_ITEM_CLEANING_FEE` from `util/types.js`
+- Add a new property to the component: `cleaningFee`
+- Add `cleaningFee` as a parameter to the `estimatedTransaction`
+  function definition and invocation.
+
+Now the `estimatedTotalPrice` function can be written as follows:
+
+```js
+const estimatedTotalPrice = (unitPrice, unitCount, cleaningFee) => {
+  const numericPrice = convertMoneyToNumber(unitPrice);
+  const cleaningFeePrice = cleaningFee
+    ? convertMoneyToNumber(cleaningFee)
+    : null;
+  const numericTotalPrice = cleaningFeePrice
+    ? new Decimal(numericPrice)
+        .times(unitCount)
+        .plus(cleaningFeePrice)
+        .toNumber()
+    : new Decimal(numericPrice).times(unitCount).toNumber();
+  return new Money(
+    convertUnitToSubUnit(
+      numericTotalPrice,
+      unitDivisor(unitPrice.currency)
+    ),
+    unitPrice.currency
+  );
+};
+```
+
+And the function invocation respectively:
+
+```js
+const totalPrice = estimatedTotalPrice(
+  unitPrice,
+  unitCount,
+  cleaningFee
+);
+```
+
+The line items list can now be resolved as follows inside the
```

Comment:
Alternatively, this could be written in a bit shorter way:

```
const cleaningFeeLineItem = {
  code: LINE_ITEM_CLEANING_FEE,
  includeFor: ['customer', 'provider'],
  unitPrice: cleaningFee,
  quantity: new Decimal(1),
  lineTotal: cleaningFee,
  reversal: false,
};

const cleaningFeeLineItemMaybe = cleaningFee ? [cleaningFeeLineItem] : [];
```
Add the `cleaningFeeLineItem` by spreading the`cleaningFeeLineItemMaybe` array to the returned `lineItems` attribute:
```
lineItems: [
  ...cleaningFeeLineItemMaybe,
  // then the current content (booking line-item)
],
```